web/src/providers/TestDefinition/TestDefinition.provider.tsx

```diff
@@ -64,14 +66,15 @@ const TestDefinitionProvider: React.FC<IProps> = ({children, testId, runId}) =>
   const isInitialized = useAppSelector(state => TestDefinitionSelectors.selectIsInitialized(state));
   const {data: test} = useGetTestByIdQuery({testId});
 
-  const {add, cancel, publish, remove, dryRun, update, isDraftMode, init, reset, revert} = useTestDefinitionCrud({
-    testId,
-    runId,
-  });
+  const {add, cancel, publish, runTest, remove, dryRun, update, isDraftMode, init, reset, revert} =
+    useTestDefinitionCrud({
+      testId,
+      runId,
+    });
 
   useEffect(() => {
-    init(run.result);
-  }, [init, isInitialized, reset, run.result]);
+    if (run.state === 'FINISHED') init(run.result);
+  }, [init, run.result, run.state]);
```

Comment:
This was causing a race condition with the `dryRun` call. 
I think we should call the init function only when the test state is finished, otherwise it was setting the `isInitialized` flag with an empty `definitionList`.